packages/jest-cli/src/TestRunner.js

```diff
@@ -123,43 +127,9 @@ class TestRunner {
 
   runTests(testPaths: Array<string>, maybeReporter?: TestReporter) {
     const config = this._config;
-    if (!maybeReporter) {
-      const TestReporter = require(config.testReporter);
-      if (config.useStderr) {
-        maybeReporter = new TestReporter(Object.create(
-          process,
-          {stdout: {value: process.stderr}},
-        ));
-      } else {
-        maybeReporter = new TestReporter();
-      }
-    }
-
-    // Prove that `reporter` exists to Flow
-    const reporter = maybeReporter;
-    if (!reporter) {
-      throw new Error('No reporter specified!');
-    }
-
     testPaths = this._sortTests(testPaths);
-
-    const aggregatedResults = {
-      didUpdate: false,
-      numFailedTests: 0,
-      numFailedTestSuites: 0,
-      numPassedTests: 0,
-      numPassedTestSuites: 0,
-      numPendingTests: 0,
-      numRuntimeErrorTestSuites: 0,
-      numTotalTests: 0,
-      numTotalTestSuites: testPaths.length,
-      snapshotFilesRemoved: 0,
-      startTime: Date.now(),
-      success: false,
-      testResults: [],
-    };
-
-    reporter.onRunStart && reporter.onRunStart(config, aggregatedResults);
+    const aggregatedResults = getEmptyAggregatedResults(testPaths.length);
+    this._reporters.forEach(r => r.onRunStart(config, aggregatedResults));
```

Comment:
please no single character variables here and below. They are confusing and hard to search for. `reporter` is preferred.